src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -305,7 +305,7 @@ protected override Encoding StandardOutputEncoding
 
         protected override int ExecuteTool(string pathToTool, string responseFileCommands, string commandLineCommands)
         {
-            if (!UseSharedCompilation || this.ToolPath != null )
+            if (!UseSharedCompilation || !String.IsNullOrEmpty(this.ToolPath))
```

Comment:
If that's the case that's fine.  Mainly i wanted to check because it does seem like a behavior difference on the surface.